tests/core/test_table_diff.py

```diff
@@ -134,3 +135,111 @@ def test_data_diff_decimals(sushi_context_fixed_date):
     )
     assert diff.row_diff().full_match_count == 2
     assert diff.row_diff().partial_match_count == 1
+
+
+@pytest.mark.slow
+def test_distinct_grain_check(sushi_context_fixed_date):
+    engine_adapter = sushi_context_fixed_date.engine_adapter
+
+    engine_adapter.ctas(
+        "table_diff_source",
+        pd.DataFrame(
+            {
+                "key": [1, 2, 3, 3, 2],
```

Comment:
your tests needs to have multiple keys and Null values

Reply:
Sure I will enhance these tests and add more